saleor/graphql/core/scalars.py

```diff
@@ -163,3 +164,19 @@ class Minute(graphene.Int):
 
 class Day(graphene.Int):
     """The `Day` scalar type represents number of days by integer value."""
+
+
+class PositiveInteger(graphene.Int):
+    """Positive Int custom field.
+
+    Should use in places where int value must be nonnegative (0 or greater).
+    """
+
+    @staticmethod
+    def parse_literal(ast):
+        value = super(PositiveInteger, PositiveInteger).parse_literal(ast)
+        if value and value < 0:
+            raise GraphQLError(
```

Comment:
We recently moved away from raising GraphQLError in scalars implementation. (ref: https://github.com/saleor/saleor/pull/14843)

Reply:
`PositiveDecimal` returns `None`, should I do the same?

Reply:
Yes.